trace_decoder/src/core.rs

```diff
@@ -107,13 +115,37 @@ pub fn entrypoint(
                 trie_roots_after: after,
                 checkpoint_state_trie_root,
                 checkpoint_consolidated_hash,
-                contract_code: contract_code
-                    .into_iter()
-                    .map(|it| (keccak_hash::keccak(&it), it))
-                    .collect(),
+                contract_code: {
+                    let initcodes =
+                        byte_code
+                            .iter()
+                            .filter_map(|nonempty_txn_bytes| -> Option<Vec<u8>> {
+                                let tx_envelope =
+                                    TxEnvelope::decode(&mut &nonempty_txn_bytes[..]).unwrap();
+                                match tx_envelope.to() {
+                                    TxKind::Create => Some(tx_envelope.input().to_vec()),
+                                    TxKind::Call(_address) => None,
+                                }
+                            });
+
+                    contract_code
+                        .into_iter()
+                        .chain(initcodes)
+                        .map(|it| (keccak_hash::keccak(&it), it))
+                        .collect()
+                },
                 block_metadata: b_meta.clone(),
                 block_hashes: b_hashes.clone(),
                 burn_addr,
+                jumpdest_table: {
+                    // Note that this causes any batch containing just a single `None` to collapse
+                    // into a `None`, which causing failover to simulating jumpdest analysis for the
+                    // whole batch. There is an optimization opportunity here.
```

Comment:
would be good to mark this improvement as `TODO` with related issue ticket

Reply:
Created issue: Simulate to get jumpdests on a per-transaction basis #653 and added todo.

Reply:
Please update the `TODO` with an owner and tracking issue

Reply:
Ping @einar-polygon